homeassistant/components/nina/__init__.py

```diff
@@ -42,13 +43,30 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     await coordinator.async_config_entry_first_refresh()
 
+    entry.async_on_unload(entry.add_update_listener(_async_update_listener))
+
     hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
 
     hass.config_entries.async_setup_platforms(entry, PLATFORMS)
 
     return True
 
 
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+async def _async_update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Handle options update."""
+    config = dict(entry.data)
+    if entry.options:
+        config.update(entry.options)
+        entry.data = MappingProxyType(config)
```

Comment:
This doesn't look right, the config entry's data + options should be updated by the config + option flows.
If you want to update config entry data instead of options in the options flow, follow this pattern:
```py
                self.hass.config_entries.async_update_entry(
                    self.config_entry, data=updated_config
                )
                return self.async_create_entry(title="", data=None)
```